### ISSUE TITLE
Since upgrading pods to 2.8.10 i get "pod not found" or "field not found"

### ISSUE THREAD

**Issue by winnersmedia** [pods-framework/pods#6462]
### Description

Since upgrading the pods plugin, my existing pods won't allow me to change their structure or save them, i get an error saying "Pod not found" or "field not found" if i try to change a field.

### Version

2.8.10

### Testing Instructions

1. update the pods plugin to the latest version.
2. Go to pods admin
3. Go to edit pods
4. click on an existing pod
5. click "save pod"

### Screenshots / Screencast

![Edit_Pod_‹_Rival_Casino_Sites_—_WordPress](https://user-images.githubusercontent.com/2738871/155168192-8796e785-494a-4dd3-b650-e85007bebfdc.jpg)

### Possible Workaround

downgrade pods to 2.7*

### Site Health Information

```text
`
### wp-core ###

version: 5.9
site_language: en_US
user_language: en_US
timezone: America/New_York
permalink: /%postname%/
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: production
user_count: 1
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /var/www/vhosts/rivalcasinosites.com/httpdocs
wordpress_size: 99.91 MB (104766202 bytes)
uploads_path: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content/uploads
uploads_size: 560.13 KB (573575 bytes)
themes_path: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content/themes
themes_size: 15.13 MB (15866675 bytes)
plugins_path: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content/plugins
plugins_size: 86.64 MB (90848486 bytes)
database_size: 25.11 MB (26329088 bytes)
total_size: 227.34 MB (238384026 bytes)

### wp-dropins (2) ###

advanced-cache.php: true
maintenance.php: true

### wp-active-theme ###

name: Rival Casino Sites (rival)
version: 1.0.0
author: Rival Casino Sites
author_website: (undefined)
parent_theme: GeneratePress (generatepress)
theme_features: core-block-patterns, widgets-block-editor, automatic-feed-links, post-thumbnails, post-formats, woocommerce, title-tag, html5, customize-selective-refresh-widgets, align-wide, responsive-embeds, editor-color-palette, custom-logo, menus, editor-styles, editor-style, widgets
theme_path: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content/themes/rival
auto_update: Enabled

### wp-parent-theme ###

name: GeneratePress (generatepress)
version: 3.1.3
author: Tom Usborne
author_website: https://tomusborne.com
theme_path: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content/themes/generatepress
auto_update: Enabled

### wp-themes-inactive (5) ###

Twenty Nineteen: version: 2.2, author: the WordPress team, Auto-updates enabled
Twenty Seventeen: version: 2.9, author: the WordPress team, Auto-updates enabled
Twenty Twenty: version: 1.9, author: the WordPress team, Auto-updates enabled
Twenty Twenty-One: version: 1.5, author: the WordPress team, Auto-updates enabled
Twenty Twenty-Two: version: 1.0, author: the WordPress team, Auto-updates enabled

### wp-plugins-active (20) ###

Admin Columns Pro: version: 5.6.4, author: AdminColumns.com, Auto-updates enabled
Admin Columns Pro - Pods: version: 1.6.1, author: AdminColumns.com, Auto-updates enabled
Admin Columns Pro - Yoast SEO: version: 1.1, author: AdminColumns.com, Auto-updates enabled
Classic Editor: version: 1.6.2, author: WordPress Contributors, Auto-updates enabled
Cloudflare: version: 4.7.0, author: Cloudflare, Inc., Auto-updates enabled
Code Snippets: version: 2.14.3, author: Code Snippets Pro, Auto-updates enabled
Collapse-O-Matic: version: 1.8.2, author: twinpictures, baden03, Auto-updates enabled
Cookie Notice & Compliance for GDPR / CCPA: version: 2.2.1, author: Hu-manity.co, Auto-updates enabled
FAQ Schema For Pages And Posts: version: 2.3.0, author: Krystian Szastok (SEO Consultant), Auto-updates enabled
GP Premium: version: 2.1.2, author: Tom Usborne, Auto-updates enabled
Insert Headers and Footers: version: 1.6.0, author: WPBeginner, Auto-updates enabled
Pods - Custom Content Types and Fields: version: 2.8.10, author: Pods Framework Team, Auto-updates enabled
Polylang Pro: version: 3.1.4, author: WP SYNTEX, Auto-updates enabled
Redirection: version: 5.2.3, author: John Godley, Auto-updates enabled
Site Kit by Google: version: 1.68.0, author: Google, Auto-updates enabled
Table of Contents Plus: version: 2106, author: Michael Tran, Auto-updates enabled
WP Config File Editor: version: 1.7.1, author: AHMeD HAMeD, Auto-updates enabled
WP Rocket: version: 3.10.8, author: WP Media, Auto-updates disabled
Yoast Duplicate Post: version: 4.4, author: Enrico Battocchi & Team Yoast, Auto-updates enabled
Yoast SEO: version: 18.2, author: Team Yoast, Auto-updates enabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1690
imagemagick_version: ImageMagick 6.9.10-86 Q16 x86_64 2020-01-13 https://imagemagick.org
imagick_version: 3.6.0
file_uploads: File uploads is turned off
post_max_size: 8M
upload_max_filesize: 16M
max_effective_size: 8 MB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 15 GB
	imagick::RESOURCETYPE_DISK: 9.2233720368548E+18
	imagick::RESOURCETYPE_FILE: 6144
	imagick::RESOURCETYPE_MAP: 15 GB
	imagick::RESOURCETYPE_MEMORY: 8 GB
	imagick::RESOURCETYPE_THREAD: 1
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, ART, ARW, AVI, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, H, HALD, HDR, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBP, WMF, WMV, WMZ, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, WebP, BMP, XPM
ghostscript_version: not available

### wp-server ###

server_architecture: Linux 4.18.0-147.8.1.el8_1.x86_64 x86_64
httpd_software: nginx/1.20.2
php_version: 7.4.27 64bit
php_sapi: fpm-fcgi
max_input_variables: 1000
time_limit: 30
memory_limit: 256M
max_input_time: 60
upload_max_filesize: 16M
php_post_max_size: 8M
curl_version: 7.61.1 OpenSSL/1.1.1k
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: true

### wp-database ###

extension: mysqli
server_version: 10.3.28-MariaDB
client_version: mysqlnd 7.4.27
max_allowed_packet: 16777216
max_connections: 151

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content
WP_PLUGIN_DIR: /var/www/vhosts/rivalcasinosites.com/httpdocs/wp-content/plugins
WP_MEMORY_LIMIT: 256M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: false
WP_DEBUG_DISPLAY: false
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: true
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: Undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

### google-site-kit ###

version: 1.68.0
php_version: 7.4.27
wp_version: 5.9
reference_url: https://www.rivalcasinosites.com
amp_mode: no
site_status: connected-site
user_status: authenticated
verification_status: verified-file
connected_user_count: 1
active_modules: site-verification, search-console, analytics, analytics-4, pagespeed-insights
required_scopes: 
	openid: ✅
	https://www.googleapis.com/auth/userinfo.profile: ✅
	https://www.googleapis.com/auth/userinfo.email: ✅
	https://www.googleapis.com/auth/siteverification: ✅
	https://www.googleapis.com/auth/webmasters: ✅
	https://www.googleapis.com/auth/analytics.readonly: ✅
capabilities: 
	googlesitekit_authenticate: ✅
	googlesitekit_setup: ✅
	googlesitekit_view_posts_insights: ✅
	googlesitekit_view_dashboard: ✅
	googlesitekit_view_module_details: ✅
	googlesitekit_manage_options: ✅
enabled_features: 
	dashboardSharing: ⭕
	ideaHubModule: ⭕
	serviceSetupV2: ⭕
	swgModule: ⭕
	userInput: ⭕
	unifiedDashboard: ✅
	zeroDataStates: ⭕
search_console_property: https://www.rivalcasinosites.com/
analytics_account_id: 1906•••••
analytics_property_id: UA-1906•••••••
analytics_profile_id: 2381•••••
analytics_use_snippet: yes
analytics_4_property_id: none
analytics_4_web_data_stream_id: none
analytics_4_measurement_id: none
analytics_4_use_snippet: yes

### pods ###

pods-server-software: nginx/1.20.2
pods-user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36
pods-session-save-path: /var/lib/php/session
pods-session-save-path-exists: No
pods-session-save-path-writable: No
pods-session-max-lifetime: 1440
pods-opcode-cache-apc: No
pods-opcode-cache-memcached: No
pods-opcode-cache-opcache: No
pods-opcode-cache-redis: No
pods-object-cache-apc: No
pods-object-cache-apcu: No
pods-object-cache-memcache: No
pods-object-cache-memcached: No
pods-object-cache-redis: Yes
pods-memory-current-usage: 15.492M
pods-memory-current-usage-real: 16.000M
pods-network-wide: No
pods-install-location: /wp-content/plugins/pods/
pods-developer: No
pods-tableless-mode: No
pods-relationship-table-enabled: Yes
pods-light-mode: No
pods-strict: No
pods-allow-deprecated: Yes
pods-api-cache: Yes
pods-shortcode-allow-evaluate-tags: No
pods-sessions: Disable sessions
pods-can-use-sessions: No
pods-relationship-table-status: 2,663

### pll_options ###

browser: 0
rewrite: 1
hide_default: 1
force_lang: 1
redirect_lang: 1
media_support: 0
sync: 0
post_types: post, page, wp_block, currency, gp_elements
taxonomies: 0
domains: 0
version: 3.1.4
default_lang: us
primary: us => 2 | uk => 17 | ca => 23 | au => 0
secondary: us => 24 | uk => 24 | ca => 24 | au => 24
previous_version: 3.1.3

### pll_language_us ###

term_id: 6
name: United States
slug: us
term_group: 0
term_taxonomy_id: 6
count: 20
tl_term_id: 7
tl_term_taxonomy_id: 7
tl_count: 1
locale: en_US
is_rtl: 0
w3c: en-US
facebook: en_US
home_url: https://www.rivalcasinosites.com/
search_url: https://www.rivalcasinosites.com/
mo_id: 285
page_on_front: 36
page_for_posts: 0
flag_code: us
flag_url: https://www.rivalcasinosites.com/wp-content/plugins/polylang-pro/vendor/wpsyntex/polylang/flags/us.png
custom_flag_url: 0
fallbacks: 0

### pll_language_uk ###

term_id: 10
name: United Kingdom
slug: uk
term_group: 1
term_taxonomy_id: 10
count: 3
tl_term_id: 11
tl_term_taxonomy_id: 11
tl_count: 1
locale: en_GB
is_rtl: 0
w3c: en-GB
facebook: en_GB
home_url: https://www.rivalcasinosites.com/uk/
search_url: https://www.rivalcasinosites.com/uk/
mo_id: 459
page_on_front: 467
page_for_posts: 0
flag_code: gb
flag_url: https://www.rivalcasinosites.com/wp-content/plugins/polylang-pro/vendor/wpsyntex/polylang/flags/gb.png
custom_flag_url: 0
fallbacks: 0

### pll_language_ie ###

term_id: 32
name: Ireland
slug: ie
term_group: 1
term_taxonomy_id: 32
count: 0
tl_term_id: 33
tl_term_taxonomy_id: 33
tl_count: 1
locale: en_IE
is_rtl: 0
w3c: en-IE
facebook: 0
home_url: https://www.rivalcasinosites.com/ie/
search_url: https://www.rivalcasinosites.com/ie/
mo_id: 848
page_on_front: 0
page_for_posts: 0
flag_code: ie
flag_url: https://www.rivalcasinosites.com/wp-content/plugins/polylang-pro/vendor/wpsyntex/polylang/flags/ie.png
custom_flag_url: 0
fallbacks: 0

### pll_language_ca ###

term_id: 18
name: Canada
slug: ca
term_group: 2
term_taxonomy_id: 18
count: 3
tl_term_id: 19
tl_term_taxonomy_id: 19
tl_count: 1
locale: en_CA
is_rtl: 0
w3c: en-CA
facebook: en_US
home_url: https://www.rivalcasinosites.com/ca/
search_url: https://www.rivalcasinosites.com/ca/
mo_id: 672
page_on_front: 673
page_for_posts: 0
flag_code: ca
flag_url: https://www.rivalcasinosites.com/wp-content/plugins/polylang-pro/vendor/wpsyntex/polylang/flags/ca.png
custom_flag_url: 0
fallbacks: 0

### pll_language_au ###

term_id: 27
name: Australia
slug: au
term_group: 3
term_taxonomy_id: 27
count: 2
tl_term_id: 28
tl_term_taxonomy_id: 28
tl_count: 1
locale: en_AU
is_rtl: 0
w3c: en-AU
facebook: en_US
home_url: https://www.rivalcasinosites.com/au/
search_url: https://www.rivalcasinosites.com/au/
mo_id: 844
page_on_front: 846
page_for_posts: 0
flag_code: au
flag_url: https://www.rivalcasinosites.com/wp-content/plugins/polylang-pro/vendor/wpsyntex/polylang/flags/au.png
custom_flag_url: 0
fallbacks: 0

### pll_language_nz ###

term_id: 37
name: New Zealand
slug: nz
term_group: 4
term_taxonomy_id: 37
count: 0
tl_term_id: 38
tl_term_taxonomy_id: 38
tl_count: 1
locale: en_NZ
is_rtl: 0
w3c: en-NZ
facebook: en_US
home_url: https://www.rivalcasinosites.com/nz/
search_url: https://www.rivalcasinosites.com/nz/
mo_id: 849
page_on_front: 0
page_for_posts: 0
flag_code: nz
flag_url: https://www.rivalcasinosites.com/wp-content/plugins/polylang-pro/vendor/wpsyntex/polylang/flags/nz.png
custom_flag_url: 0
fallbacks: 0

`
```

**Comment by JoryHogeveen** [comment]
Hi @winnersmedia 

This is most likely a specific issue with your installation or it might be cache. Did you try clearing your cache?

**Comment by winnersmedia** [comment]
i've cleared the cache for wp rocket and the pods cache but the issue still seems to happen.

**Comment by theham** [comment]
We are experiencing the same issue on our end. 

After upgrading the pods_api to load a pod just returns 'Pod Not Found'.

Cache and everything have been cleared multiple times.

I have run through all the 2.X versions.

Rolling back to 2.7.31 fixes the issue.

**Comment by anobjectn** [comment]
I did a rollback a few days ago to 2.8.13 because a pods page was redirecting. 
Expected:  "something/*"
Instead: get request for "something/xyz" redirected to "/xyz" as if the "something" segment of the path was not there.

I do not know if its the same issue as OP but in my case, reverting the plugin brought back expected redirect behavior.

**Comment by anobjectn** [comment]
In hopes this is helpful, in my case my pod page routes fine up to version 2.8.13 - once I install version 2..8.14 , the pod page is improperly redirected (just like the latest tested: 2.8.16)

**Comment by anobjectn** [comment]
if I had to guess, I would say maybe a change in pods/components/Pages.php as there are changes involving tests of slugs, if a pod exists, and the pods_pages_auto_404 filter ... as one of the symptoms I see in my site is that route change I mentioned in my first comment results in a 404 (unless a legacy posts at the misdirected path once existed). Newest version where no issue for me is 2.8.13 and I see the issue in 2.8.14 as well as 2.8.16.

**Comment by sc0ttkclark** [comment]
This may have been resolved in Pods 2.9.12, leaving it open until we get confirmation after the release

**Comment by sc0ttkclark** [comment]
Closing as fixed, we can reopen if needed
